### DISCUSSION TITLE
Check empty results in collection loop

### DISCUSSION THREAD

**Discussion by ronvdc** [statamic/cms#5102] [Q&A]
I have a collection of tour dates, each of which consist of a date and a description. I want to show a list of tours, but not tours in the past. I use this code for that, and it works great!

```
{{ collection:tours sort="slug:asc" }}
   {{ if tour_date | is_today || tour_date | is_future }}
      <p><i class="bx bx-chevron-right"></i> <b>{{ tour_date format="l, F j, Y" }}</b>{{ if tour_comment }}, {{ tour_comment }}.{{ endif }}</p>
   {{ endif }}
{{ /collection:tours }}
```

If there are no tour dates to show (because all entries in the collections have dates in the past, how do I show a message instead of nothing? I read something about no_results, but I can't seem to figure out how to code this. 

I tried something like:

```
{{ collection:tours sort="slug:asc" }}
	{{ if no_results }}
		<p>There are no scheduled tours at the moment. Please check back later for the latest tour schedule.</p>
	{{ else }}
		{{ if tour_date | is_today || tour_date | is_future }}
			<p><i class="bx bx-chevron-right"></i> <b>{{ tour_date format="l, F j, Y" }}</b>{{ if tour_comment }}, {{ tour_comment }}.{{ endif }}</p>
                {{ endif }}
	{{ endif }}
{{ /collection:tours }}
```

That doesn't show the message, but I'm guessing that's because 'no_results' is for the collection as a whole, not just tour dates in the  future, so even if all entries have tour dates in the past, the 'no_results' condition is never met as there are results (i.e., entries in the 'tours' collection, just not the ones I want to show). I'm blanking out here now.
Does anyone have any suggestions?

Thanks so much!
--Ron

**Comment by ryanmitchell** [comment]
You could try to use conditions on the actual collections tag: https://statamic.dev/conditions#string-conditions, which then means you have limited the results so no_results should be valid.

`{{ collection:tours sort="slug:asc" :tour_date:is_after="now" }}...{{ /collection:tours }}`

`now` is a variable Statamic defines: https://statamic.dev/variables/now

It won't handle your is_today condition, but it gets you most of the way there which is maybe enough of a compromise?
